src/components/MessagingView.tsx

```diff
@@ -4,333 +4,943 @@ import {FhirClientContext, FhirClientContextType} from "../FhirClientContext";
     }
 
-    private saveMessage() {
-        // @ts-ignore
-        let context: FhirClientContextType = this.context;
-        let newMessage: CommunicationRequest = CommunicationRequest.createNewManualOutgoingMessage(this.state.activeMessage, context.patient, context.carePlan);
-        console.log("Attempting to save new CommunicationRequest:", newMessage);
-        context.client.create(newMessage).then(
-            (savedCommunicationRequest: IResource) => {
-                console.log("Saved new CommunicationRequest:", savedCommunicationRequest);
-                this.state.temporaryCommunications.unshift(Communication.tempCommunicationFrom(savedCommunicationRequest));
-                this.setState({
-                    activeMessage: "",
-                    temporaryCommunications: this.state.temporaryCommunications
-                });
-            },
-            (reason: any) => {
-                console.log("Failed to create new CommunicationRequest:", reason);
-                this.setState({error: reason});
-            }
-        ).catch((reason: any) => {
-            console.log("Failed to create new CommunicationRequest:", reason);
-            this.setState({error: reason});
-        });
+    if (!context.carePlan || !this.state.communications) {
+      return <CircularProgress />;
     }
 
-    private _buildMessageRow(message: Communication, index: number): React.ReactNode {
-        let incoming = true;
-        if (message.recipient && message.recipient.find((r: IReference) => r.reference.includes("Patient"))) {
-            incoming = false;
-        }
-
-        let timestamp = null;
-        let delivered = true;
-        if (message.sent) {
-          let datetime = new Date(message.sent);
-          // instead of hard-coded en-US possibly get locale info from config?
-          timestamp = `${datetime.toLocaleDateString("en-US", {
-            weekday: "short",
-            year: "numeric",
-            month: "numeric",
-            day: "numeric",
-          })} ${datetime.toLocaleTimeString()}`;
-        } else {
-            delivered = false;
-        }
-        // let timestamp = `${intl.formatDate(datetime)} ${intl.formatTime(datetime)}`;
-        let msg = message.displayText();
-        let autoMessage = true;
-        if (!message.category) {
-            console.log("Communication is missing category");
-        } else if (message.category.find((c: ICodeableConcept) => c.coding.find((coding: ICoding) => IsaccMessageCategory.isaccManuallySentMessage.equals(coding)))) {
-            autoMessage = false;
-        }
-        let bubbleStyle = MessagingView.getBubbleStyle(incoming, autoMessage, delivered);
-        let priority = message.priority;
-        let themes: string[] = message.getThemes();
-        return this._alignedRow(incoming, msg, timestamp, bubbleStyle, priority, index, themes);
-
+    let messageBoxProps = {
+      maxHeight: 548,
```

Comment:
It was originally [600](https://github.com/uwcirg/isacc-messaging-client-sof/blob/main/src/components/MessagingView.tsx#L48)  I reduced it to try to fit the messages list and the bottom widget for entering message to fit within the viewport.  I have since changed it back to 600, as you are right, it is an arbitrary number and there is probably a better way to specify this.